### ISSUE TITLE
NLP Capstone project - In all models val_loss trains to nan (SimpleRNN at least)

### ISSUE THREAD

**Issue by letyrodridc** [udacity/aind-issue-reports#214]
Many students were reporting this problem. 
When they train the simple model, they get **val_loss = nan**, for example:

Scroll right ->
```
Train on 110288 samples, validate on 27573 samples
Epoch 1/10
110288/110288 [==============================] - 6s 58us/step - loss: 2.8908 - acc: 0.4301 - val_loss: nan - val_acc: 0.5258
Epoch 2/10
110288/110288 [==============================] - 4s 40us/step - loss: 2.0730 - acc: 0.5435 - val_loss: nan - val_acc: 0.5518
Epoch 3/10
110288/110288 [==============================] - 4s 40us/step - loss: 1.7677 - acc: 0.5733 - val_loss: nan - val_acc: 0.5821
Epoch 4/10
110288/110288 [==============================] - 4s 40us/step - loss: 1.6108 - acc: 0.5912 - val_loss: nan - val_acc: 0.5994
Epoch 5/10
110288/110288 [==============================] - 4s 39us/step - loss: 1.5151 - acc: 0.6036 - val_loss: nan - val_acc: 0.6093
Epoch 6/10
110288/110288 [==============================] - 4s 40us/step - loss: 1.4394 - acc: 0.6162 - val_loss: nan - val_acc: 0.6201
Epoch 7/10
110288/110288 [==============================] - 4s 40us/step - loss: 1.3780 - acc: 0.6277 - val_loss: nan - val_acc: 0.6336
Epoch 8/10
110288/110288 [==============================] - 4s 41us/step - loss: 1.3311 - acc: 0.6343 - val_loss: nan - val_acc: 0.6365
Epoch 9/10
110288/110288 [==============================] - 4s 41us/step - loss: 1.2935 - acc: 0.6403 - val_loss: nan - val_acc: 0.6399
Epoch 10/10
110288/110288 [==============================] - 4s 41us/step - loss: 1.2625 - acc: 0.6452 - val_loss: nan - val_acc: 0.6476
la jersey est parfois chaud en l' et il est est en en <PAD> <PAD> <PAD> <PAD> <PAD> <PAD> <PAD> <PAD>
```

The proposed solution is:
> I solved this by passing in len(french_tokenizer.word_index) + 1 as the french_vocab_size:
simple_rnn_model = simple_model(
    tmp_x.shape,
    preproc_french_sentences.shape[1],
    len(english_tokenizer.word_index) + 1,
    len(french_tokenizer.word_index) + 1)
The tokenizer generates indices from 1 to 344/345, but then padding the sequences introduces index 0, so the one hot tensor needs to be at least 345/346 for sparse_categorical_crossentropy to work properly.

Link: https://discussions.udacity.com/t/capstone-project-model-1-rnn-implementation-invalidargumenterror-received-label-value-344-outside-valid-range-0-344/296300/12?u=chogall

Maybe some comment could be added to the notebook and change the rubric so reviewers check that val_loss is not nan.

This occurs training all the models.

**Comment by letyrodridc** [comment]
Confirmed, it's a bug in the project code. 
This is a comparison against reviewers training version:
![image](https://user-images.githubusercontent.com/9123307/36080176-eb0dae90-0f6a-11e8-8391-51d04049a6e9.png)

the +1 is missing in the repository:
https://github.com/udacity/aind2-nlp-capstone/blob/master/machine_translation.ipynb

@cgearhart This bug is from the beginning in the Machine Translation project code.

**Comment by letyrodridc** [comment]
Tentative Solution: 

Replace
```
english_vocab_size = len(english_tokenizer.word_index)
french_vocab_size = len(french_tokenizer.word_index)
```
 with
```
# Includes the padding word
english_vocab_size = len(english_tokenizer.word_index)+1
french_vocab_size = len(french_tokenizer.word_index)+1
```
in the notebook.

The solution needs to be tested.
